src/db/index.js

```diff
@@ -19,16 +19,31 @@ const db = {
  * Be sure to wrap each migration script in a try/catch and throw on Error.
  * We don't want the server to run if the migration failed.
  */
-async function applyMigrations() {
+async function applyMigrations(db) {
   // Run migration scripts here
+  const migrationPath = require('path').join(__dirname, 'migrations');
+  const migrations = [];
+  try {
+    await fs.readdirSync(migrationPath).forEach(async (file) => {
+      if(file.endsWith('.js')){
+        const migration = await require(`./migrations/${file}`);
+        migrations.push(migration);
+      }
+    });
+  } catch (err) {
+    throw Error(`Migration scripts load Error ${err.message}`);
+  }
+
+  // execute sequentially
+  migrations.reduce((promise, currentTask) => promise.then(() => currentTask(db)), Promise.resolve()).catch((err) => {
```

Comment:
convert to async await syntax

Reply:
should be using `map` or `forEach`. 

reduce is meant to run functionality and add to an accumulator: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce